coalib/testing/GlobalBearTestHelper.py

```diff
@@ -0,0 +1,66 @@
+import os.path
+
+from contextlib import contextmanager
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.testing.BaseTestHelper import BaseTestHelper
+
+
+@contextmanager
+def execute_bear(bear, *args, **kwargs):
+    try:
+        bear_output_generator = bear.execute(*args, **kwargs)
+        assert (bear_output_generator is not None
+                ), 'Bear returned None on execution\n'
+        yield bear_output_generator
+    except Exception as err:
+        msg = []
+        while not bear.message_queue.empty():
+            msg.append(bear.message_queue.get().message)
+        raise AssertionError(str(err) + ' \n' + '\n'.join(msg))
+    return list(bear_output_generator)
+
+
+def get_test_path(test_dir,
+                  file):
+    return os.path.join(os.path.dirname(__file__),
+                        test_dir, file)
+
+
+def get_results(global_bear,
+                test_dir,
+                file_dict,
+                settings={}):
+    for file in file_dict:
+        with get_test_path(test_dir, file) as fname, \
+            execute_bear(global_bear, fname, **settings
+                         ) as bear_output:
+            return bear_output
+
+
+class GlobalBearTestHelper(BaseTestHelper):
+    """
+    """
+
+    def check_results(self,
+                      global_bear,
+                      results,
+                      filename=None,
+                      settings={}):
+
+        assert isinstance(self, BaseTestHelper)
+        assert isinstance(
```

Comment:
use Python 3 annotations on the method arguments, and we have a decorator which will automatically do type checking of them.